streams/src/main/java/org/apache/kafka/streams/state/internals/ExceptionUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import java.util.LinkedList;
+
+final class ExceptionUtils {
+    private ExceptionUtils() {}
+
+    static LinkedList<RuntimeException> executeAll(final Runnable... actions) {
+        LinkedList<RuntimeException> suppressed = null;
+        for (final Runnable action : actions) {
+            try {
+                action.run();
+            } catch (final RuntimeException exception) {
+                if (suppressed == null) {
+                    suppressed = new LinkedList<>();
+                }
+                suppressed.add(exception);
+            }
+        }
+        return suppressed;
+    }
+
+    static void throwSuppressed(final String message, final LinkedList<RuntimeException> suppressed) {
+        if (suppressed != null && !suppressed.isEmpty()) {
+            final RuntimeException firstCause = suppressed.pollFirst();
+            final RuntimeException toThrow = new RuntimeException(message, firstCause);
+            for (final RuntimeException e : suppressed) {
+                toThrow.addSuppressed(e);
```

Comment:
The rest of the exceptions are listed as "suppressed by" the exception we're throwing.